Fix jaeger container docs (#53572)
quarkusio/quarkus: docs/src/main/asciidoc/opentelemetry-tracing.adoc

=== Create the configuration

:opentelemetry-config:
include::{includes}/opentelemetry-config.adoc[]

== See data

> Before starting the app, please setup the system to visualize the OpenTelemetry data.

Reviewer: ```suggestion
Before starting the app, please set up the system to visualize the OpenTelemetry data.
```

Author: done

We have several options:

> * Start an all-in-one Grafana OTel LGTM dev service for traces, logs and metrics.

Reviewer: ```suggestion
* Start an all-in-one Grafana OTel LGTM Dev Service for traces, logs and metrics.
```

Author: doce

> * Jaeger system just for traces.

Reviewer: ```suggestion
* Jaeger system just for traces
```

Author: done

* Logging exporter

=== Grafana OTel LGTM option
A Dev service that will receive your app's telemetry. Only the dependency is needed.

* Take a look at: xref:observability-devservices-lgtm.adoc[Getting Started with Grafana-OTel-LGTM].

This features a Quarkus Dev service including a Grafana for visualizing data, Loki to store logs, Tempo to store traces and Prometheus to store metrics. Also provides an OTel collector to receive the data.

=== Jaeger v2 to see traces option

> Jaeger V2 is a tool to visualize span, and it's based on OpenTelemetry collector. There is no need to install a separated collector.

Reviewer: ```suggestion
Jaeger V2 is a tool to visualize spans, and it's based on OpenTelemetry collector. There is no need to install a separate collector.
```

Author: done

More details are available in https://medium.com/jaegertracing/towards-jaeger-v2-moar-opentelemetry-2f8239bee48e[this blog post].

Start the OpenTelemetry Collector and Jaeger system via the following Docker command:

[source,shell]
----
docker run -it \
  -p 16686:16686 \
  -p 4317:4317 \
  -p 4318:4318 \
jaegertracing/jaeger:latest
----
Where:
|===
|Port | Purpose

|16686
|Jaeger UI

|4317
|OpenTelemetry collector. OTLP protobuf gRPC receiver.

|4318
|OpenTelemetry collector. OTLP protobuf HTTP receiver.

|===

Other ports are available in the https://www.jaegertracing.io/docs/2.17/architecture/apis/#write-apis[Jaeger documentation].

=== Logging exporter

> You can output all traces to the console by setting the exporter to logging in the application.properties file:

Reviewer: ```suggestion
You can output all traces to the console by setting the exporter to `logging` in the `application.properties` file:
```

Author: done

[source,properties]
----
quarkus.otel.traces.exporter=logging

> ----

Reviewer: Can you add new lines between the paragraphs before/after and the source block: some Asciidoc engines struggle with this...

Author: Sure. Will keep that in mind

> Set the exporter to logging. Normally you don’t need to set this property. The default is cdi.

Reviewer: The `Set the exporter to logging.` is odd as you're repeating yourself? Did you mean it to be a callout?

Author: I've re-wrote that part for clarity.

Also add this dependency to your project:
[source,xml]
----
<dependency>
<groupId>io.opentelemetry</groupId>

> <artifactId>opentelemetry-exporter-logging</artifactId>

Reviewer: ```suggestion
    <groupId>io.opentelemetry</groupId>
    <artifactId>opentelemetry-exporter-logging</artifactId>
```

Author: done
